.travis/before_install.sh

```diff
@@ -0,0 +1,116 @@
+  echo "Invalid Required PR link detected in commit message. Please use the full https url."
+  exit 1
+fi
+
+if [ "$TRAVIS_PULL_REQUEST" != "false" ] || [ -z "$TRAVIS_TAG" -a "$TRAVIS_BRANCH" != "master"]
+then
+  export PULP_PR_NUMBER=$(echo $COMMIT_MSG | grep -oP 'Required\ PR:\ https\:\/\/github\.com\/pulp\/pulpcore\/pull\/(\d+)' | awk -F'/' '{print $7}')
+  export PULP_SMASH_PR_NUMBER=$(echo $COMMIT_MSG | grep -oP 'Required\ PR:\ https\:\/\/github\.com\/pulp\/pulp-smash\/pull\/(\d+)' | awk -F'/' '{print $7}')
+  export PULP_ROLES_PR_NUMBER=$(echo $COMMIT_MSG | grep -oP 'Required\ PR:\ https\:\/\/github\.com\/pulp\/ansible-pulp\/pull\/(\d+)' | awk -F'/' '{print $7}')
+  export PULP_BINDINGS_PR_NUMBER=$(echo $COMMIT_MSG | grep -oP 'Required\ PR:\ https\:\/\/github\.com\/pulp\/pulp-openapi-generator\/pull\/(\d+)' | awk -F'/' '{print $7}')
+  export PULP_OPERATOR_PR_NUMBER=$(echo $COMMIT_MSG | grep -oP 'Required\ PR:\ https\:\/\/github\.com\/pulp\/pulp-operator\/pull\/(\d+)' | awk -F'/' '{print $7}')
+else
+  export PULP_PR_NUMBER=
+  export PULP_SMASH_PR_NUMBER=
+  export PULP_ROLES_PR_NUMBER=
+  export PULP_BINDINGS_PR_NUMBER=
+  export PULP_OPERATOR_PR_NUMBER=
+fi
+
+# dev_requirements contains tools needed for flake8, etc.
+# So install them here rather than in install.sh
+pip install -r dev_requirements.txt
+
+# check the commit message
+# ./.travis/check_commit.sh
+
+cd ../pulp_catdog
+# run black separately from flake8 to get a diff
+black --diff --check --exclude docs/conf.py .
+
+# Lint code.
+#flake8 --config flake8.cfg
+
+cd ..
+git clone --depth=1 https://github.com/pulp/ansible-pulp.git
+if [ -n "$PULP_ROLES_PR_NUMBER" ]; then
+  cd ansible-pulp
+  git fetch --depth=1 origin +refs/pull/$PULP_ROLES_PR_NUMBER/merge
+  git checkout FETCH_HEAD
+  cd ..
+fi
+
+
+git clone --depth=1 https://github.com/pulp/pulp-operator.git
+if [ -n "$PULP_OPERATOR_PR_NUMBER" ]; then
+  cd pulp-operator
+  git fetch --depth=1 origin +refs/pull/$PULP_OPERATOR_PR_NUMBER/merge
+  git checkout FETCH_HEAD
+  RELEASE_VERSION=v0.9.0
+  curl -LO https://github.com/operator-framework/operator-sdk/releases/download/${RELEASE_VERSION}/operator-sdk-${RELEASE_VERSION}-x86_64-linux-gnu
+  chmod +x operator-sdk-${RELEASE_VERSION}-x86_64-linux-gnu && sudo mkdir -p /usr/local/bin/ && sudo cp operator-sdk-${RELEASE_VERSION}-x86_64-linux-gnu /usr/local/bin/operator-sdk && rm operator-sdk-${RELEASE_VERSION}-x86_64-linux-gnu
+  sudo operator-sdk build --image-builder=docker quay.io/pulp/pulp-operator:latest
+  cd ..
+fi
+
+
+git clone --depth=1 https://github.com/pulp/pulpcore.git
+
+if [ -n "$PULP_PR_NUMBER" ]; then
+  cd pulpcore
+  git fetch --depth=1 origin +refs/pull/$PULP_PR_NUMBER/merge
+  git checkout FETCH_HEAD
+  cd ..
+fi
+
+rm ./pulpcore/containers/images/pulp/container-assets/pulp-api
+mv ./plugin_template/.travis/pulp-api ./pulpcore/containers/images/pulp/container-assets/pulp-api
```

Comment:
So why are we replacing this?

Reply:
for running `django-admin makemigrations catdog`
https://github.com/pulp/plugin_template/blob/master/.travis/pulp-api#L14